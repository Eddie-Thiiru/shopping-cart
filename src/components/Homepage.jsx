import { useState, useEffect } from "react";
import PropTypes from "prop-types";
import About from "./About";
import "../styles/Homepage.css";
import png from "../../jason.png";

const Adventure = ({ background }) => {
  return (
    <div className="adventureSection">
      <h2>Action-Adventure Games</h2>
      <div className="sectionImage">
        <img src={png} alt="" />
      </div>
    </div>
  );
};

const Action = ({ background }) => {
  return (
    <div className="actionSection">
      <div className="sectionImage">
        <img src={background} alt="" />
      </div>
      <div className="sectionDescription">
        <h2>Action Games</h2>
        <p></p>
      </div>
    </div>
  );
};

const RPG = ({ background }) => {
  return (
    <div className="rpgSection">
      <div className="sectionImage">
        <img src={background} alt="" />
      </div>
      <div className="sectionDescription">
        <h2>RPG Games</h2>
        <p></p>
      </div>
    </div>
  );
};

const Strategy = ({ background }) => {
  return (
    <div className="strategySection">
      <div className="sectionImage">
        <img src={background} alt="" />
      </div>
      <div className="sectionDescription">
        <h2>Strategy Games</h2>
        <p></p>
      </div>
    </div>
  );
};

const Platform = ({ background }) => {
  return (
    <div className="platformSection">
      <div className="sectionImage">
        <img src={background} alt="" />
      </div>
      <div className="sectionDescription">
        <h2>Platform Games</h2>
        <p></p>
      </div>
    </div>
  );
};

const Racing = ({ background }) => {
  return (
    <div className="racingSection">
      <div className="sectionImage">
        <img src={background} alt="" />
      </div>
      <div className="sectionDescription">
        <h2>Racing Games</h2>
        <p></p>
      </div>
    </div>
  );
};

const Sports = ({ background }) => {
  return (
    <div className="sportsSection">
      <div className="sectionImage">
        <img src={background} alt="" />
      </div>
      <div className="sectionDescription">
        <h2>Sports Games</h2>
        <p></p>
      </div>
    </div>
  );
};

const Group = ({ children }) => {
  return <div className="groupSection">{children}</div>;
};

const Homepage = () => {
  const [data, setData] = useState({
    actionURL: "",
    adventureURL: "",
    rpgURL: "",
    strategyURL: "",
    platformURL: "",
    racingURL: "",
    sportsURL: "",
  });
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await fetch();
        // " https://api.rawg.io/api/games/3498?key=ded6236dee784a49946f45a75db16ec8",
        // { mode: "cors" },
        const data = await response.json();

        if (!data.error) {
          console.log(data);
          setData(data.background_image);
        } else {
          throw new Error(data.error);
        }
      } catch (error) {
        setError(true);
      } finally {
        setLoading(false);
      }
    };

    fetchData();
  }, []);

  return (
    <div className="homepage">
      <About />
      <Adventure />
      <Group>
        <Action background={data.actionURL} />
        <RPG background={data.rpgURL} />
        <Strategy background={data.strategyURL} />
        <Platform background={data.platformURL} />
        <Racing background={data.racingURL} />
        <Sports background={data.sportsURL} />
      </Group>
    </div>
  );
};

Adventure.propTypes = {
  background: PropTypes.string,
};

Action.propTypes = {
  background: PropTypes.string,
};

RPG.propTypes = {
  background: PropTypes.string,
};

Strategy.propTypes = {
  background: PropTypes.string,
};

Platform.propTypes = {
  background: PropTypes.string,
};

Racing.propTypes = {
  background: PropTypes.string,
};

Sports.propTypes = {
  background: PropTypes.string,
};

Group.propTypes = {
  children: PropTypes.array,
};

export default Homepage;
