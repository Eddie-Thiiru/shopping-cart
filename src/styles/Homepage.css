.homepage {
  display: grid;
  grid-template-rows: 850px 850px 850px;
}

.adventureSection {
  display: grid;
  grid-template-rows: 50px 1fr;
  justify-items: center;
  align-items: center;
}

.groupSection {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr 1fr;
}

.actionSection,
.rpgSection,
.strategySection,
.platformSection,
.racingSection,
.sportsSection {
  display: grid;
  grid-template-rows: 1fr 100px;
}

.adventureSection > div {
  height: 90%;
  width: 80%;
}

.adventureSection img {
  max-width: 100%;
  min-height: 100%;
  object-fit: cover;
  /* filter: opacity(0.5); */
}
