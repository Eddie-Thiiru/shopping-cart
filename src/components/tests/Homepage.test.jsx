import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Homepage from "../Homepage";

// test sections are rendered
// test images are renderd
// test descriptions are rendered

describe("Homepage", () => {
  it("renders homepage component", () => {
    const { container } = render(<Homepage />);

    expect(container).toMatchSnapshot();
  });

  it("should render all sections' headings", () => {
    render(<Homepage />);

    screen.debug();

    expect(screen.getByText(/video game store/i)).toBeInTheDocument();
    expect(screen.getByText(/action games/i)).toBeInTheDocument();
    expect(screen.getByText(/adventure games/i)).toBeInTheDocument();
    expect(screen.getByText(/rpg games/i)).toBeInTheDocument();
    expect(screen.getByText(/strategy games/i)).toBeInTheDocument();
    expect(screen.getByText(/platform games/i)).toBeInTheDocument();
    expect(screen.getByText(/racing games/i)).toBeInTheDocument();
    expect(screen.getByText(/sports games/i)).toBeInTheDocument();
  });
});

// vi.mock("../Homepage", () => {
//   return {
//     default: {
//       getFetchData: vi.fn(async () => {
//         try {
//           const response = await fetch("https://product-api.com/api/fakeurl");
//           const data = await response.json();

//           if (!data.error) {
//             return data;
//           } else {
//             throw new Error(data.error);
//           }
//         } catch (error) {
//           return null;
//         }
//       }),
//     },
//   };
// });
