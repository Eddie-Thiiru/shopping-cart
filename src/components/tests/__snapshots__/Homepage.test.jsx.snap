// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > renders homepage component 1`] = `
<div>
  <div
    class="homepage"
  >
    <div
      class="aboutSection"
    >
      <div
        class="aboutTitle"
      >
        <h2>
          Welcome
        </h2>
        <h2>
          Video Game Store
        </h2>
      </div>
      <p>
        Browse thousands of games in our collection, from action packed games to the. You want it? We got it!
      </p>
      <div>
        <h3>
          Game Collection
        </h3>
        <p>
          10,000+
        </p>
      </div>
      <div
        class="platforms"
      >
        <div
          class="nintendo"
        >
          <img />
          <p>
            Nintendo
          </p>
        </div>
        <div
          class="playstation"
        >
          <img />
          <div>
            PlayStation
          </div>
        </div>
        <div
          class="xbox"
        >
          <img />
          <div>
            XBOX
          </div>
        </div>
        <div
          class="pc"
        >
          <img />
          <div>
            PC
          </div>
        </div>
      </div>
    </div>
    <div
      class="adventureSection"
    >
      <h2>
        Action-Adventure Games
      </h2>
      <div
        class="sectionImage"
      >
        <img
          alt=""
          src="/jason.png"
        />
      </div>
    </div>
    <div
      class="groupSection"
    >
      <div
        class="actionSection"
      >
        <div
          class="sectionImage"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="sectionDescription"
        >
          <h2>
            Action Games
          </h2>
          <p />
        </div>
      </div>
      <div
        class="rpgSection"
      >
        <div
          class="sectionImage"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="sectionDescription"
        >
          <h2>
            RPG Games
          </h2>
          <p />
        </div>
      </div>
      <div
        class="strategySection"
      >
        <div
          class="sectionImage"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="sectionDescription"
        >
          <h2>
            Strategy Games
          </h2>
          <p />
        </div>
      </div>
      <div
        class="platformSection"
      >
        <div
          class="sectionImage"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="sectionDescription"
        >
          <h2>
            Platform Games
          </h2>
          <p />
        </div>
      </div>
      <div
        class="racingSection"
      >
        <div
          class="sectionImage"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="sectionDescription"
        >
          <h2>
            Racing Games
          </h2>
          <p />
        </div>
      </div>
      <div
        class="sportsSection"
      >
        <div
          class="sectionImage"
        >
          <img
            alt=""
            src=""
          />
        </div>
        <div
          class="sectionDescription"
        >
          <h2>
            Sports Games
          </h2>
          <p />
        </div>
      </div>
    </div>
  </div>
</div>
`;
